sympy/physics/mechanics/_geometry.py

```diff
@@ -0,0 +1,571 @@
+"""Implementations of geometry objects for use by wrapping pathways.
+
+Notes
+=====
+
+This module is experimental and so is named with a leading underscore to
+indicate that the API is not yet stabilized and could be subject to breaking
+changes.
+
+"""
+
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import TYPE_CHECKING
+
+from sympy.core.backend import Integer, Mul, acos, pi, sqrt, tan
+from sympy.functions.elementary.trigonometric import atan2
+from sympy.polys.polytools import cancel
+from sympy.simplify.simplify import posify, trigsimp
+
+if TYPE_CHECKING:
+    from sympy.core.backend import Symbol
+    from sympy.core.expr import Expr
+    from sympy.physics.mechanics import Point, Vector
+
+
+__all__ = [
+    'Cylinder',
+    'Sphere',
+]
+
+
+class GeometryBase(ABC):
+    """Abstract base class for all geometry classes to inherit from.
+
+    Notes
+    =====
+
+    Instances of this class cannot be directly instantiated by users. However,
+    it can be used to created custom geometry types through subclassing.
+
+    """
+
+    @abstractmethod
+    def _point_is_on_surface(self, point: Point) -> bool:
+        """Determine if a point is on the geometry's surface.
+
+        Parameters
+        ==========
+        point : Point
+            The point for which it's to be ascertained if it's on the
+            geometry's surface or not.
+
+        """
+
```

Comment:
Does this need "pass" here?

Reply:
Good spot. It doesn't strictly need `pass` as a docstring acts as a sufficient method body for Python to correctly parse the file, but I think it's better style to explicitly mark the empty method body for abstract methods. Addressed in https://github.com/sympy/sympy/pull/25068/commits/603783485b7b2a3cbeb21737a4154436ea95629c.